server/src/main/kotlin/suwayomi/tachidesk/manga/impl/backup/proto/ProtoBackupExport.kt

```diff
@@ -30,11 +34,107 @@ import suwayomi.tachidesk.manga.model.table.MangaStatus
 import suwayomi.tachidesk.manga.model.table.MangaTable
 import suwayomi.tachidesk.manga.model.table.SourceTable
 import suwayomi.tachidesk.manga.model.table.toDataClass
+import suwayomi.tachidesk.server.ApplicationDirs
+import suwayomi.tachidesk.server.serverConfig
 import java.io.ByteArrayOutputStream
+import java.io.File
 import java.io.InputStream
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Timer
+import java.util.TimerTask
 import java.util.concurrent.TimeUnit
+import java.util.prefs.Preferences
 
 object ProtoBackupExport : ProtoBackupBase() {
+    private val logger = KotlinLogging.logger { }
+    private val applicationDirs by DI.global.instance<ApplicationDirs>()
+    private const val lastAutomatedBackupKey = "lastAutomatedBackupKey"
+    private val preferences = Preferences.userNodeForPackage(ProtoBackupExport::class.java)
+
+    private val backupTimer = Timer()
+    private var currentAutomatedBackupTask: TimerTask? = null
+
+    fun scheduleAutomatedBackupTask() {
+        if (!serverConfig.automatedBackups) {
+            currentAutomatedBackupTask?.cancel()
+            return
+        }
+
+        val minInterval = TimeUnit.DAYS.toMillis(1L)
+        val intervalInMS = TimeUnit.DAYS.toMillis(serverConfig.backupInterval.toLong())
+        val backupInterval = intervalInMS.coerceAtLeast(minInterval)
+
+        val lastAutomatedBackup = preferences.getLong(lastAutomatedBackupKey, 0)
+        val initialDelay = backupInterval - (System.currentTimeMillis() - lastAutomatedBackup) % backupInterval
+
+        currentAutomatedBackupTask?.cancel()
+        currentAutomatedBackupTask = object : TimerTask() {
+            override fun run() {
+                cleanupAutomatedBackups()
+                createAutomatedBackup()
+                preferences.putLong(lastAutomatedBackupKey, System.currentTimeMillis())
+            }
+        }
+
+        backupTimer.scheduleAtFixedRate(currentAutomatedBackupTask, initialDelay, backupInterval)
+    }
+
+    private fun createAutomatedBackup() {
+        logger.info { "Creating automated backup..." }
+
+        val inputStream = createBackup(
+            BackupFlags(
+                includeManga = true,
+                includeCategories = true,
+                includeChapters = true,
+                includeTracking = true,
+                includeHistory = true
+            )
+        )
+
+        val automatedBackupDir = File(applicationDirs.automatedBackupRoot)
+        automatedBackupDir.mkdirs()
+
+        val backupFile = File(applicationDirs.automatedBackupRoot, getBackupFilename())
+
+        inputStream.use { input -> backupFile.outputStream().use { output -> input.copyTo(output) } }
```

Comment:
Just to make sure the inputStream gets closed if there is an error creating the file.
```suggestion
        createBackup(
            BackupFlags(
                includeManga = true,
                includeCategories = true,
                includeChapters = true,
                includeTracking = true,
                includeHistory = true
            )
        ).use { input ->
            val automatedBackupDir = File(applicationDirs.automatedBackupRoot)
            automatedBackupDir.mkdirs()

            val backupFile = File(applicationDirs.automatedBackupRoot, getBackupFilename())

            backupFile.outputStream().use { output -> input.copyTo(output) }
        }
```